akka-stream/src/main/scala/akka/stream/impl/io/FilePublisher.scala

```diff
@@ -52,6 +52,7 @@ private[akka] final class FilePublisher(f: Path, completionPromise: Promise[IORe
       chan = FileChannel.open(f, FilePublisher.Read)
     } catch {
       case ex: Exception ⇒
+        completionPromise.success(IOResult(0L, Failure(ex)))
```

Comment:
OK, I'll change to `NonFatal` and also use `trySuccess` to make the code more robust to future changes.